pkg/agent/sysadvisor/plugin/poweraware/capper/server/service.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+
+	"github.com/kubewharf/katalyst-core/pkg/agent/qrm-plugins/advisorsvc"
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/poweraware/capper"
+	"github.com/kubewharf/katalyst-core/pkg/config"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+)
+
+const (
+	// ServiceNamePowerCap also is the unix socket name of the server is listening on
+	ServiceNamePowerCap = "node_power_cap"
+
+	metricPowerCappingTargetName           = "power_capping_target"
+	metricPowerCappingResetName            = "power_capping_reset"
+	metricPowerCappingNoActorName          = "power_capping_no_actor"
+	metricPowerCappingLWSendResponseFailed = "power_capping_lw_send_response_failed"
+)
+
+type powerCapService struct {
+	sync.Mutex
+	capInstruction *capper.CapInstruction
+	notify         *fanoutNotifier
+	emitter        metrics.MetricEmitter
+	grpcServer     *grpcServer
+}
+
+func (p *powerCapService) Start() error {
```

Comment:
Let's add the corresponding `Stop` method to stop the server when the poweraware plugin shuts down.

Reply:
will add Stop method and call it when the plugin exits